lib/util/buffered_readable.js

```diff
@@ -31,7 +31,7 @@ BufferedReadable.prototype._read = function() {
   if (this._buffer.length > 0) {
     for (var i = 0; i < this._buffer.length; i++) {
       if (!this.push(this._buffer[i])) {
-        this._buffer = this._buffer.slice(i);
+        this._buffer = this._buffer.slice(i + 1);
```

Comment:
According to the docs "push" returns whether or not we should keep pushing data. Internally the ReadableState has its own buffer as well.